### ISSUE TITLE
is there a way to use Quantum without AudioEngine?

### ISSUE THREAD

**Issue by designerfuzzi** [Ableton/link#57]
as i am writing Quartz Composer API implementation came around the need of Quantum setting without the use of AudioEngine. 
Quartz Composer Compositions are basically producing visual content only controlled in three different time spaces (none, idle, extern/local) with an macro traversing engine. So its similar to pureData or maxmsp but not same. 
So could rewrite code using EngineData structure without Audiobuffers but it would be easier when using the link without metronome but quantum would be supported. Or i need a hint what i didnt see yet.

**Comment by designerfuzzi** [comment]
ah forgot something. Why is setting of Quantum without AudioEngine interesting?
Because different Macro patches or different parallel running patches can have in-depended quantum settings and still follow the same sessionstate.tempo() but dont need layering of sounds and multiple running audiobuffer callbacks at all. 

As plugins in QuartzComposer are literally same as APPs on iOS with the difference having a predefined QCView to render into instead of UIView, allocated plugins should run independent. Was considering and testing a singleton class but turned out i would have to write a Quantum interpreter that is running in each of the instances on top of the Link quantum. Thought i can avoid doubling functions.

Independent quantum per instance has flavor that a visual can be prepared running in a 1 beat quantum while another one is looping in 4 or different. Substructures would follow that timescale where they are wired/connected to. I also could just turn off AudioEngine always after allocating one link peer to ignore metronomes sound but the audiobuffers are still generated and consume CPU power. Is there a way of make use of Quantum processing without AudioBuffers?

**Comment by fgo-ableton** [comment]
The Quantum is not part of the Link object but a parameter when querying phase/time info. So it is possible to share a Link instance between the different parts of your app/plugin and query information for different Quanta at different locations in your code.
The audio engine is of cause not required when not calculating audio. You can simply use the clock provided by Link to get the current time: Ie. `mLink.clock().micros()` Depending on the latency of your system it has to be taken into account as well. So you maybe need to shift that time a little depending on when you calculate the image and when it is visible.
Make sure to use the right functions to capture/commit the SessionState. The `...App...` functions are thread safe. The `...Audio...` functions are lockfree.

**Comment by designerfuzzi** [comment]
okay got it working with independent Quanta and managed to turn off use off AudioEngine for this special case. Any patch connected to one placed Link "producer" runs as its own Link in the whole session, just same as an iOS app does. So different Patches running in parallel appear as their own Link app. Thanks for the hint.
